src/app/public/modules/modal/modal.service.ts

```diff
@@ -1,23 +1,32 @@
 import {
-  ApplicationRef,
-  ComponentFactoryResolver,
   ComponentRef,
   Injectable
 } from '@angular/core';
 
-import { SkyModalInstance } from './modal-instance';
-import { SkyModalHostComponent } from './modal-host.component';
-import { SkyModalAdapterService } from './modal-adapter.service';
-import { SkyModalConfigurationInterface as IConfig } from './modal.interface';
+import {
+  SkyDynamicComponentService
+} from '@skyux/core';
+
+import {
+  SkyModalInstance
+} from './modal-instance';
+
+import {
+  SkyModalHostComponent
+} from './modal-host.component';
+
+import {
+  SkyModalConfigurationInterface as IConfig
+} from './modal.interface';
 
 @Injectable()
 export class SkyModalService {
   private static host: ComponentRef<SkyModalHostComponent>;
 
+  // TODO: In future breaking change - remove extra parameters as they are no longer used.
+  /* tslint:disable:no-unused-variable */
   constructor(
-    private resolver: ComponentFactoryResolver,
-    private appRef: ApplicationRef,
-    private adapter: SkyModalAdapterService
+    private dynamicComponentService?: SkyDynamicComponentService
```

Comment:
What you're doing here is correct, but I'm curious if we can move this service to the adapter so that you wouldn't have to deprecate the `addHostEl` and `removeHostEl` methods? You could call the dynamic component service methods in there instead?

Reply:
We can't...... The `SkyDynamicComponentService` requires `Injector`. This is only available when you inject the service into a component. If I inject `SkyDynamicComponentService` into the adapter I get an `undefined provider` error. I ran into the same thing when I attempted to inject `Injector` into the adapter in an earlier version of this work.

Reply:
That's right. 

In that case, just remove the adapter methods that are no longer used. I appreciate the effort to reduce the negative effects of this refactor, but in this case we're not "deprecating" the method; they literally don't work anymore (if they were deprecated, they'd still work, but wouldn't be actively supported). Also, these methods are attached to an internal implementation, so the negative effects should be minimal.